shaptools/hana.py

```diff
@@ -99,6 +97,80 @@ def is_installed(self):
             self._logger.error(err)
             return False
 
+    @classmethod
+    def update_conf_file(cls, conf_file, **kwargs):
+        """
+        Update config file parameters
+
+        Args:
+            conf_file (str): Path to the configuration file
+            kwargs (opt): Dictionary with the values to be updated.
+                Use the exact name of the SAP configuration file for the key
+
+        kwargs can be used in the next two modes:
+            update_conf_file(conf_file, sid='PRD', hostname='hana01')
+            update_conf_file(conf_file, **{'sid': 'PRD', 'hostname': 'hana01'})
+        """
```

Comment:
Maybe it would be a good idea to handle the case where the key is not already set in the configuration file?

Reply:
@krig yes, this option is contamplated. This method updates any parameter that you want, you just need to pass it using kwargs, with the exact same key used in the config files. If you don't want, you don't need to update the sid, for example. This is totally up to the user. The logic about the parameters is created in the salt modules and states, not here

Reply:
Right, but if the pattern is never found in the file, it doesn't add the new_value line, right? Or am I missing that part..

Reply:
If the pattern is not found, the file won't be changed, it will keep the same content. I think that this is a quite safe option.
Yes, we expect a correct usage of the patterns, otherwise, at least the changes to add some syntax error in the file are really small

Reply:
Well, usually configuration values have defaults so that if they aren't set in the file, they get the default value. So a user could edit the file and delete the entry for a value and expect the default to get used. Which would then break this code, right?

Reply:
So if the pattern is not found, but the user sets a value for this key, in that case the file is not updated, you are right. If the pattern is not found, the file keeps as it is.

I will update the code to consider the option that the pattern is not found, to add this value to the end of the file

Reply:
@krig I have done the updates. The code adds a new line with the new value.
Unfortunately, i think that i will be useless...
HANA configuration files are separated in segments, and each segment has a header ([SERVER] for the sid for example). So basically, if we just add the lines at the bottom of the file, is not going to work (unless we put the header too, and if it exists, it fails...).

In this case, I think that the best will be just to let the code as it is, and add this information in the docstring, saying that a correct conf file should be used and that it can be create using *create_conf_file* method.

What do you think?

Reply:
Hmm right, yes.. we would need to know which section to put the value in...
Actually, don't we need to know that anyway? Because if there are sections, maybe the same key could appear multiple times in different sections. So then the current code would rewrite all of them.

Reply:
According the parameter reference (https://help.sap.com/viewer/2c1988d620e04368aa4103bf26f17727/2.0.03/en-US/c16432a77b6144dcb75aace2b4fcacff.html), there is not any duplicated entry, so the risk is lower.